src/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,116 @@
+import { Payment } from './PaymentCard';
+import { NetworkInfo, CryptoAsset } from '@windingtree/win-commons/dist/types';
+import { useState, useMemo, useCallback, useEffect } from 'react';
+import {
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  FormControlLabel,
+  Radio
+} from '@mui/material';
+import { BigNumber, utils } from 'ethers';
+import { getNetworkInfo } from '../config';
+import Logger from '../utils/logger';
+
+const logger = Logger('CurrencySelector');
+
+export interface CheckoutPrice {
+  value: BigNumber;
+  currency: string;
+}
+
+export interface PriceSelectProps {
+  payment?: Payment;
+  network?: NetworkInfo;
+  onQuote(useQuote: boolean): void;
+}
+
+export const CurrencySelector = ({ payment, network, onQuote }: PriceSelectProps) => {
+  const [value, setValue] = useState<string | null>(payment ? payment.currency : null);
```

Comment:
maybe you can make this more declartive, as `value` is quit generic. e.g. `currency and `setCurrency`

Reply:
this variable has the name `value` because this variable is used as a value of the radio button block. But, you are right - it will be better to call this variable - `currency`, I will rename it.

Reply:
renamed to `currency`